src/Compilers/CSharp/Portable/Parser/SyntaxParser.cs

```diff
@@ -474,6 +474,22 @@ protected SyntaxToken EatToken(SyntaxKind kind)
             return CreateMissingToken(kind, this.CurrentToken.Kind, reportError: true);
         }
 
+        // Consume a token if it is the right kind. Otherwise skip a token and replace it with one of the correct kind.
```

Comment:
Suggestion:
`// Eat and return the current token, but wrapped it as a missing token if it is not the expected kind.`